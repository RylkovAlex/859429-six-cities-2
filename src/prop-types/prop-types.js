import PropTypes from 'prop-types';

export const offerCardPropTypes = {
  city: PropTypes.shape({
    name: PropTypes.string.isRequired,
    location: PropTypes.shape({
      latitide: PropTypes.number,
      longitude: PropTypes.number,
      zoom: PropTypes.number,
    }),
  }),
  id: PropTypes.number.isRequired,
  previewImage: PropTypes.string.isRequired,
  images: PropTypes.arrayOf(PropTypes.string),
  title: PropTypes.string.isRequired,
  isFavorite: PropTypes.bool.isRequired,
  isPremium: PropTypes.bool.isRequired,
  rating: PropTypes.number.isRequired,
  type: PropTypes.string.isRequired,
  bedrooms: PropTypes.number.isRequired,
  maxAdults: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  goods: PropTypes.arrayOf(PropTypes.string),
  host: PropTypes.shape({
    id: PropTypes.number.isRequired,
    isPro: PropTypes.bool,
    name: PropTypes.stringisRequired,
    avatarUrl: PropTypes.string,
  }),
  description: PropTypes.string,
  location: PropTypes.shape({
    latitide: PropTypes.number,
    longitude: PropTypes.number,
    zoom: PropTypes.number,
  }),
};

export const reviewPropTypes = {
  id: PropTypes.number.isRequired,
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    isPro: PropTypes.bool.isRequired,
    name: PropTypes.string.isRequired,
    avatarUrl: PropTypes.string.isRequired,
  }),
  rating: PropTypes.number.isRequired,
  comment: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
};

export const cityPropTypes = {
  name: PropTypes.string.isRequired,
  location: PropTypes.shape({
    latitude: PropTypes.number.isRequired,
    longitude: PropTypes.number.isRequired,
    zoom: PropTypes.number.isRequired,
  }),
};

export const offerCardForTests = {
  city: {
    name: ``,
    location: {
      latitude: 1,
      longitude: 1,
      zoom: 1,
    },
  },
  id: 0,
  previewImage: ``,
  images: [``],
  title: ``,
  isFavorite: false,
  isPremium: false,
  rating: 0,
  type: ``,
  bedrooms: 0,
  maxAdults: 0,
  price: 0,
  goods: [``],
  host: {
    id: 0,
    isPro: false,
    name: ``,
    avatarUrl: ``,
  },
  description: ``,
  location: {
    latitude: 1,
    longitude: 1,
    zoom: 1,
  },
};

export const reviewForTests = {
  id: 0,
  user: {
    id: 0,
    isPro: false,
    name: ``,
    avatarUrl: ``,
  },
  rating: 0,
  comment: ``,
  date: `Thu Nov 07 2019 13:56:54 GMT+0300 (Москва, стандартное время)`,
};
