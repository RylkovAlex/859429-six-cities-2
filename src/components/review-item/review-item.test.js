import React from 'react';
import renderer from 'react-test-renderer';
import ReviewItem from './review-item';
import {reviewForTests} from '../../prop-types/prop-types';

it(`ReviewItem correctly renders after relaunch`, () => {
  const props = {
    review: reviewForTests,
  };

  const tree = renderer
    .create(<ReviewItem {...props}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
