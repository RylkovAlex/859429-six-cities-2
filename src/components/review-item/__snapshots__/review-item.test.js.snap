// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewItem correctly renders after relaunch 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height="54"
        src=""
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "NaN",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      
    </p>
    <time
      className="reviews__time"
      dateTime="2019-11-07T10:50:41.000Z"
    >
      Wed Nov 07 2001 00:00:00 GMT+0300 (GMT+03:00)
    </time>
  </div>
</li>
`;
